src/Parser/Hunter/BeastMastery/Modules/Items/Tier20_2p.js

```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+
+import SPELLS from 'common/SPELLS/index';
+
+import Analyzer from 'Parser/Core/Analyzer';
+import Combatants from 'Parser/Core/Modules/Combatants';
+import { formatNumber } from 'common/format';
+import getDamageBonus from "Parser/Hunter/Shared/Core/getDamageBonus";
+import SpellIcon from "common/SpellIcon";
+import ITEMS from "common/ITEMS/HUNTER";
+import SpellLink from "common/SpellLink";
+
+const T20_2P_MODIFIER_PR_STACK = 0.015;
+
+const debug = false;
+
+class Tier20_2p extends Analyzer {
+  static dependencies = {
+    combatants: Combatants,
+  };
+
+  bonusDmg = 0;
+  bestialWrathBaseModifier = 0;
+  currentStacks = 0;
+
+  on_initialized() {
+    this.active = this.combatants.selected.hasBuff(SPELLS.HUNTER_BM_T20_2P_BONUS.id);
+    if (this.combatants.selected.hasTalent(SPELLS.BESTIAL_FURY_TALENT) || this.combatants.selected.hasFinger(ITEMS.SOUL_OF_THE_HUNTMASTER.id)) {
+      this.bestialWrathBaseModifier = 0.4;
+    } else {
+      this.bestialWrathBaseModifier = 0.25;
+    }
+  }
+
+  on_byPlayer_cast(event) {
+    const spellId = event.ability.guid;
+    if (spellId === SPELLS.BESTIAL_WRATH.id) {
+      this.currentStacks = 0;
+      debug && console.log('Bestial Wrath Cast and currentStacks = ', this.currentStacks);
+    }
+    if (spellId !== SPELLS.COBRA_SHOT.id && spellId !== SPELLS.KILL_COMMAND.id && spellId !== SPELLS.MULTISHOT.id) {
+      return;
+    }
+    if (this.combatants.selected.hasBuff(SPELLS.BESTIAL_WRATH.id)) {
+      this.currentStacks += 1;
+      debug && console.log('t20 buffer cast and stacks = ', this.currentStacks);
+    }
+  }
+
+  on_byPlayer_damage(event) {
+    if (!this.combatants.selected.hasBuff(SPELLS.BESTIAL_WRATH.id, event.timestamp)) {
+      return;
+    }
+    this.bonusDmg += getDamageBonus(event, (1 - ((1 / (1 + this.bestialWrathBaseModifier)) * (1 + (this.currentStacks * T20_2P_MODIFIER_PR_STACK)))));
```

Comment:
Pretty sure this calculation is incorrect (see my previous calculation comment). Pretty sure the correct calculation is `getDamageBonus(event, (this.currentStacks * T20_2P_MODIFIER_PR_STACK) / (1 + this.bestialWrathBaseModifier));`. Remember it's "newMultiplier / oldMultiplier === relativeMultiplier"

Reply:
It needs to be ``getDamageBonus(event, (1+this.bestialWrathBaseModifier+(this.currentStacks * T20_2P_MODIFIER_PR_STACK)) / (1 + this.bestialWrathBaseModifier));`` surely then?

Reply:
Not quite, it would be that `- 1`. However, what I put it is the same thing in a reduced form. Sorry I should have mentioned I reduced it.